webui/react/src/hooks/useSettingsProvider.tsx

```diff
@@ -42,17 +42,17 @@ export const SettingsProvider: React.FC<React.PropsWithChildren> = ({ children }
   });
   const checked = useObservable(authChecked);
   const [canceler] = useState(new AbortController());
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading] = useState(() => observable(true));
   const querySettings = useRef('');
-  const [settingsState, setSettingsState] = useState(() => Map<string, Settings>());
+  const [settingsState] = useState(() => observable(Map<string, Settings>()));
```

Comment:
question: how come we need to wrap values with `observable`? to match the props type? it looks unnatural to me.
 @ClaireNeveu Is it the right way to use `micro-observable`?

Reply:
This is correct usage. This value is an observable and we avoid intermediate re-renders that we saw with this being normal react state thereby improving performance.